temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -384,6 +387,12 @@ private void handleSingleEventLookahead(HistoryEvent event) {
           }
           flags.setSdkFlag(sdkFlag);
         }
+        if (!Strings.isNullOrEmpty(completedEvent.getSdkMetadata().getSdkName())) {
```

Comment:
Pardon my lack of Java proto knowledge, can you confirm (like some other lang proto impls) that a getter for a message returns an empty form of the message even if the underlying value is null?

Reply:
Yes it does